File: Domain/Organization.cs
using Ecotrack_Api.Domain.Common;


namespace Ecotrack_Api.Domain;


public class Organization : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Country { get; set; }
    public string? Sector { get; set; }
    public Settings Settings { get; set; } = new();
}


public class Settings
{
    public string Timezone { get; set; } = "America/Tegucigalpa";
    public string Currency { get; set; } = "HNL";
}

File: models/Organization.js
// models/Organization.js
const { Schema, model, Types } = require('mongoose');

const organizationSchema = new Schema({
    name: { type: String, required: true, trim: true },
    country: { type: String, trim: true },
    sector: { type: String, trim: true },
    settings: {
        timezone: { type: String, default: 'America/Tegucigalpa' },
        currency: { type: String, default: 'HNL' }
    }
}, { timestamps: true });

organizationSchema.index({ name: 1 });

module.exports = model('Organization', organizationSchema);
